CFG.md

```diff
@@ -0,0 +1,55 @@
+# Sonar-PHP Control Flow Graph
+
+----
+## Specification
+
+A CFG is a graph of [basic blocks](https://en.wikipedia.org/wiki/Basic_block) (from now on, we will use the term 'block').
+
+### Blocks
+
+A block has the following attributes:
+
+* successors - blocks that are executed after
+* predecessors - blocks that are executed before
+* syntactic successor - if no jump is applied, the block that immediately follows
+* elements - the AST Trees that are inside the block
```

Comment:
How are these elements stored? Is this a list of elements?

Reply:
Usually a basic block ends with a terminator -- if this is not the case, please document that as it is usually assumed.